packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -52,6 +79,39 @@ export class VaDate {
   dateBlur: EventEmitter;
 
   private handleDateBlur = (event: FocusEvent) => {
+    const [year, month, day] = (this.value || '').split('-').map(val => val);
+    const yearNum = parseInt(year);
+    const monthNum = parseInt(month);
+    const dayNum = parseInt(day);
+    // Use a leading zero for numbers < 10
+    const numFormatter = new Intl.NumberFormat('en-US', {
+      minimumIntegerDigits: 2,
+    });
+    this.value = `${year}-${month ? numFormatter.format(monthNum) : ''}-${
+      day ? numFormatter.format(dayNum) : ''
+    }`;
+    const daysForSelectedMonth = monthNum > 0 ? days[monthNum] : [];
+    const leapYear = checkLeapYear(yearNum);
+    // Check validity of date if invalid provide message and error state styling
+    if (
+      yearNum < minYear ||
+      yearNum > maxYear ||
+      monthNum < minMonths ||
+      monthNum > maxMonths ||
+      dayNum < minMonths ||
+      dayNum > daysForSelectedMonth.length ||
+      day === '' ||
+      month === '' ||
+      year === '' ||
```

Comment:
If we're calling `parseInt` on the year, month, & day anyway, why not do it in the `.map()` function? I see that the non-parsed values are used down near the bottom of this if statement, but if we were dealing with `NaN` I think that something like `!!year` could work